src/bun.js/bindings/JSBuffer.cpp

```diff
@@ -995,20 +1004,142 @@ static inline JSC::EncodedJSValue jsBufferPrototypeFunction_fillBody(JSC::JSGlob
     }
 }
 
-static inline JSC::EncodedJSValue jsBufferPrototypeFunction_includesBody(JSC::JSGlobalObject* lexicalGlobalObject, JSC::CallFrame* callFrame, typename IDLOperation<JSBuffer>::ClassParameter castedThis)
+static int indexOf(const uint8_t* thisPtr, size_t thisLength, const uint8_t* valuePtr, size_t valueLength, int byteOffset)
+{
+    if (thisLength < valueLength + byteOffset)
+        return -1;
+    auto start = thisPtr + byteOffset;
+    auto end = thisPtr + thisLength;
+    auto it = std::search(start, end, std::boyer_moore_searcher(valuePtr, valuePtr + valueLength));
+    if (it != end) {
+        return it - thisPtr;
+    }
+    return -1;
+}
+
+static int lastIndexOf(const uint8_t* thisPtr, size_t thisLength, const uint8_t* valuePtr, size_t valueLength, int byteOffset)
+{
+    auto start = thisPtr;
+    auto end = thisPtr + std::min(thisLength, byteOffset + valueLength);
+    auto it = std::find_end(start, end, valuePtr, valuePtr + valueLength);
```

Comment:
trying to find a libc version of this that doesn't need a terminating `\0`

Reply:
I don't think there is one. Otherwise someone will mention here :p
https://stackoverflow.com/questions/12274339/is-there-a-function-implement-reverse-memmem-in-c-or-c